geps/gep-3779/index.md

```diff
@@ -42,20 +37,432 @@ Kubernetes core provides NetworkPolicies as one way to do it. Network Policies h
+  Kind Kind `json:"kind"`
+
+  // Selector is the label selector of target objects of the specified kind.
+  Selector *metav1.LabelSelector `json:"selector"`
+}
+
+```
+
+##### Enhanced Discoverability with `gwctl`
+
+A key challenge with `LabelSelector` is the loss of discoverability. It’s easier to see which policies target a `Service` but difficult to determine which policies might affect a specific pod.
+
+To address this, **investment in tooling is required.** Specifically, the `gwctl` CLI tool should be enhanced to provide insights such as:
+
+```sh
+gwctl describe pods pod1 
+
+...
+InheritedPolicies:
+  Type                   Name                                 Target Kind   Target Name/Expression
+  ----                   ----                                 -----------   -----------
+  AuthorizationPolicy  demo-authz-policy-on-payment-pods      Pod           Selector={foo: a, bar: 2}
+...
+
+# List all AuthorizationPolicy resources in json format in the active namespace
+gwctl get authorizationPolicy -o json
+
+```
+
+Without dedicated tooling, the `LabelSelector` approach could significantly degrade the user experience and observability.
+
+##### Namespace-Wide Policies
+
+A common case for authorization is to target a whole namespace. We can achieve the above with a few patterns:
+
+###### Option 1
+
+Target a namespace, name MUST be empty (as this is already namespaced policy)
+
+```yaml
+targetRefs:
+- Kind: Namespace
+```
+
+This however gives the (confusing) impression that Gateways (whether in-cluster or off-cluster gateways) are targeted as well. Beyond the question of "does it really make sense for an E/W policy to be applicable to a N/S Gateway?" it leaves no way for users to do a namespace wide policy that excludes Gateways (no way to exclude Off-cluster Gateways with labels -- see option 3 for alternative).
+
+###### Option 2
+
+Leaving an empty targetRefs. While this is theoretically possible without a breaking change, this would be another fundamental change to policy attachment (to allow policies without a targetRef).
+Additionally, this is also suffer from the same problem of being perceived as an applicable scope for both Gateways and Workloads in the namespace.
+
+This option is also inconsistent with other API fields where an empty field or absence of a it does not mean select all. See [recent comment](https://github.com/kubernetes-sigs/gateway-api/pull/3887#discussion_r2176125600) on the proposal for empty ParentRef field.
+
+###### Option 3 (Recommended)
+
+An empty pod selector. Kubernetes official docs clarify that the semantics of empty selectors are the decision of the API owner. In fact, many Kubernetes APIs (I know Service API does the opposite :/) using empty selectors as a select-all mechanism. See [NetworkPolicy podSelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.33/#networkpolicy-v1-networking-k8s-io), PodDisruptionBudget, ResourceQuota, and more.
```

Comment:
Re-asking a question from several rounds ago: so are you suggesting that targeting a namespace means targeting all Pods in that namespace? or something else?

(WRT Gateways, note that it _is_ meaningful in many cases for a workload in a cluster to send a request to a Gateway in the same cluster, so it is definitely meaningful to talk about E/W auth when talking to the Gateway. If you view the Gateway as a point where you cross an administrative boundary, it's not even necessarily all that weird.)